src/filesystem/implementation.js

```diff
@@ -1678,6 +1678,31 @@ function mkdir(fs, context, path, mode, callback) {
   make_directory(context, path, callback);
 }
 
+function mkdtemp(fs, context, prefix, options, callback) { 
+  callback = arguments[arguments.length - 1];
+
+  // this function is used to generate a random character set to append 
+  // to tmp dir name to make sure it is unique
+  function generateRandom() {
+    return 'xxxxxx'.replace(/[xy]/g, function(c) {
+      var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
+      return v.toString(16);
+    });
+  }
+  if (prefix) {
+    var path = '/'+prefix+generateRandom();
```

Comment:
Use `path.join()` for this.  Also, you can call into `shared.js` now and use `randomChars()`